specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/stable/2020-03-01/outputs.json

```diff
@@ -0,0 +1,1066 @@
+          "description": "The name of the Azure SQL database. Required on PUT (CreateOrReplace) requests."
+        },
+        "table": {
+          "type": "string",
+          "description": "The name of the table in the Azure SQL database. Required on PUT (CreateOrReplace) requests."
+        },
+        "user": {
+          "type": "string",
+          "description": "The user name that will be used to connect to the Azure SQL database. Required on PUT (CreateOrReplace) requests."
+        },
+        "password": {
+          "type": "string",
+          "description": "The password that will be used to connect to the Azure SQL database. Required on PUT (CreateOrReplace) requests."
+        }
+      }
+    },
+    "DocumentDbOutputDataSource": {
+      "description": "Describes a DocumentDB output data source.",
+      "x-ms-discriminator-value": "Microsoft.Storage/DocumentDB",
+      "allOf": [
+        {
+          "$ref": "#/definitions/OutputDataSource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/DocumentDbOutputDataSourceProperties",
+          "description": "The properties that are associated with a DocumentDB output. Required on PUT (CreateOrReplace) requests.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "DocumentDbOutputDataSourceProperties": {
+      "description": "The properties that are associated with a DocumentDB output.",
+      "properties": {
+        "accountId": {
+          "type": "string",
+          "description": "The DocumentDB account name or ID. Required on PUT (CreateOrReplace) requests."
+        },
+        "accountKey": {
+          "type": "string",
+          "description": "The account key for the DocumentDB account. Required on PUT (CreateOrReplace) requests."
+        },
+        "database": {
+          "type": "string",
+          "description": "The name of the DocumentDB database. Required on PUT (CreateOrReplace) requests."
+        },
+        "collectionNamePattern": {
+          "type": "string",
+          "description": "The collection name pattern for the collections to be used. The collection name format can be constructed using the optional {partition} token, where partitions start from 0. See the DocumentDB section of https://docs.microsoft.com/en-us/rest/api/streamanalytics/stream-analytics-output for more information. Required on PUT (CreateOrReplace) requests."
+        },
+        "partitionKey": {
+          "type": "string",
+          "description": "The name of the field in output events used to specify the key for partitioning output across collections. If 'collectionNamePattern' contains the {partition} token, this property is required to be specified."
+        },
+        "documentId": {
+          "type": "string",
+          "description": "The name of the field in output events used to specify the primary key which insert or update operations are based on."
+        }
+      }
+    },
```

Comment:
It looks like the Microsoft.AzureFunction discriminator schema has been lost in moving forward from 2017-04-01: "x-ms-discriminator-value": "Microsoft.AzureFunction". That would be another breaking change. Was this intentional?

Reply:
Yes, this was intentional. We do not want to support Azure Functions in this version.

Reply:
I think it would be more appropriate to review from the perspective of the previous stable API version as that is what customers are expected to upgrade from.

There are things in preview that we may have introduced that we decided to not move forward with moving to stable.